rmgpy/cantherm/statmech.py

```diff
@@ -778,6 +792,41 @@ def applyEnergyCorrections(E0, modelChemistry, atoms, bonds,
 
     return E0
 
+class Log(object):
+    """
+    Represent a general log file.
+    The attribute `path` refers to the location on disk of the log file of interest.
+    A method is provided to determine whether it is a Gaussian, Molpro, or QChem type.
+    """
+
+    def __init__(self, path):
+        self.path = path
+
+    def determine_qm_software(self, fullpath):
```

Comment:
I like the suggestion, but I think it's not straight forward to implement. Since the new `__init__` will need to read the log file for software determination, the folder or full path should be passed as well. In the current implementation the user can still just pass the file name.
If I misunderstood something, let's discuss it.

Reply:
Can't you just define self.path = os.path.join(os.getcwd(),path) in __init__?

Reply:
I don't think so, I have examples where the species.py file, which defines the Log file name, is in a different folder than Cantherm's input file, so `getcwd()` won't do the trick